### PR TITLE
Implementation of linearized_directed constructor

### PR DESCRIPTION
**Pull request by d-schindler**
This PR is to provide linearized MS for directed networks as discussed in #69

### CODE REVIEW

**Comment by d-schindler** [comment]
We need to set $\alpha=1$ (no teleportation) as a default for large graphs because otherwise the quality functions is a dense matrix, which leads to memory errors.

**Comment by d-schindler** [comment]
We should have `constructors_kwargs` as an argument in `run()` such that one can change teleportation conveniently.

**Comment by d-schindler** [comment]
@arnaudon , I think this is ready to merge after your improvements. could you e.g. provide a kwarg in run to set the teleportation parameter alpha?

**Comment by arnaudon** [comment]
Here you are!

**Review by arnaudon — APPROVED** [review]

**Comment by arnaudon** [comment]
Let me know if it's ok, and we can merge!

**Comment by d-schindler** [comment]
Do you keep the print statements on purpose? Also the `print("alpha1")`?

File: src/pygenstability/constructors.py [hunk 1]
```diff
@@ -297,3 +300,59 @@ def get_data(self, scale):
...
+        """Prepare the constructor with non-scale dependent computations."""
+        alpha = kwargs.get("alpha", 0.8)
+        n_nodes = self.graph.shape[0]
+
+        out_degrees = np.array(self.graph.sum(1)).flatten()
+        _check_total_degree(out_degrees)
+        dinv = np.divide(1, out_degrees, where=out_degrees != 0)
+
+        if alpha < 1:
+            print("alpha")
```
Comment (arnaudon) [inline src/pygenstability/constructors.py:332]:
```suggestion
```

File: src/pygenstability/constructors.py [hunk 2]
```diff
@@ -297,3 +300,59 @@ def get_data(self, scale):
...
+            self.partial_quality_matrix = sp.csr_matrix(
+                alpha * np.diag(dinv).dot(self.graph.toarray())
+                + ((1 - alpha) * np.diag(np.ones(n_nodes)) + np.diag(alpha * (dinv == 0.0))).dot(
+                    ones
+                )
+                - np.eye(n_nodes)
+            )
+
+        if alpha == 1:
+            print("alpha1")
```
Comment (arnaudon) [inline src/pygenstability/constructors.py:344]:
```suggestion
```

**Comment by arnaudon** [comment]
Oups, I just removed them!

**Comment by arnaudon** [comment]
is it good to merge like that @d-schindler ?

**Comment by d-schindler** [comment]
Yes, I think it's fine now!

**Comment by d-schindler** [comment]
Before you merge, could you delete this from benchmarking.py, it's deprecated: `with_spectral_decomp=True`

**Pull request merged**